replicate/common/generic-replicator.go

```diff
@@ -354,7 +354,7 @@ func (r *GenericReplicator) replicateResourceToNamespaces(obj interface{}, targe
 	cacheKey := MustGetKey(obj)
 
 	for _, namespace := range targets {
-		if err := r.UpdateFuncs.ReplicateObjectTo(obj, &namespace); err != nil {
+		if err = r.UpdateFuncs.ReplicateObjectTo(obj, &namespace); err != nil {
 			err = multierror.Append(errors.Wrapf(err, "Failed to replicate %s %s -> %s: %v",
 				r.Kind, cacheKey, namespace.Name, err,
 			))
```

Comment:
I was thinking about this (I should have already noticed this in my earlier review -- sorry I didn't 🙄🙏):

`err` is supposed to be multi-error that contains and aggregates any individual errors that might have occurred when replicating a resource to any of the `target` namespaces. However, now `err` is overwritten with the result from `ReplicateToObject` in every iteration. The multi-error previously stored in `err` will be discarded on each iteration.

In effect, this will mean that this function will return a multi-error that contains _only_ the return value from the last invocation of `ReplicateToObject`; any errors from namespaces other than the last one will now _not_ return any kind of error.

To avoid further confusion, I'd suggest using an entirely different variable name for the inner error.

```suggestion
		if innerErr := r.UpdateFuncs.ReplicateObjectTo(obj, &namespace); innerErr != nil {
			err = multierror.Append(err, errors.Wrapf(innerErr, "Failed to replicate %s %s -> %s: %v",
				r.Kind, cacheKey, namespace.Name, innerErr,
			))
```

Also (unrelated/has always been this way), `multierror.Append` is invoked entirely incorrectly -- the first parameter needs to be the multi-error that the new error should be appended to.